cosmo_tester/test_suites/ha/ha_distributed_test.py

```diff
@@ -181,6 +181,80 @@ def test_distributed_installation_sanity(distributed_installation,
     distributed_nodecellar.uninstall()
 
 
+cluster_to_cluster_versions = [
+    '5.0',
+    'master'
+]
+cluster_upgrade_tests_runs_params = []
+for version in cluster_to_cluster_versions:
+    cluster_upgrade_tests_runs_params.append({
+        'cluster': True,
+        'cluster_upgrade': True,
+        'cluster_version': version
+    })
+
+
+@pytest.mark.parametrize('distributed_installation',
+                         cluster_upgrade_tests_runs_params,
+                         indirect=True)
+def test_cluster_to_cluster_upgrade(distributed_installation,
+                                    cfy,
+                                    logger,
+                                    tmpdir,
+                                    attributes,
+                                    distributed_nodecellar):
+    logger.info('Running Sanity check for cluster with an external database')
+    rabbitmq = distributed_installation.message_queue
+
+    manager1 = distributed_installation.manager
+    old_manager_1 = distributed_installation.old_cluster[0]
+
+    old_manager_1.use()
+
+    create_and_add_user_to_tenant(cfy, logger)
+
+    set_sanity_user(cfy, old_manager_1, logger)
+
+    # Creating secrets with 'tenant' visibility
+    create_secrets(cfy, logger, attributes, old_manager_1)
+
+    distributed_nodecellar.upload_and_verify_install()
+
+    set_admin_user(cfy, old_manager_1, logger)
+
+    # Create and download a snapshot
+    snapshot_id = 'SNAPSHOT_ID'
+    local_snapshot_path = str(tmpdir / 'snap.zip')
+    logger.info('Creating snapshot')
+    create_snapshot(old_manager_1, snapshot_id, attributes, logger)
+    download_snapshot(old_manager_1, local_snapshot_path, snapshot_id, logger)
+
+    set_admin_user(cfy, manager1, logger)
+
+    # Upload and restore snapshot to the external AIO manager
+    logger.info('Uploading and restoring snapshot')
+    upload_snapshot(manager1, local_snapshot_path, snapshot_id, logger)
+    restore_snapshot(manager1, snapshot_id, cfy, logger,
+                     change_manager_password=False)
+    time.sleep(7)
```

Comment:
magic number, is there a wait for exe func? maybe

Reply:
Inside `restore_snapshot` it actually waits for the execution to finish.
This is just a few seconds to wait for the services to start-up again.